src/sagemaker/local/image.py

```diff
@@ -120,9 +120,16 @@ def train(self, input_data_config, hyperparameters):
         shutil.rmtree(data_dir)
         # Also free the container config files.
         for host in self.hosts:
-            shutil.rmtree(os.path.join(self.container_root, host))
+            container_config_path = os.path.join(self.container_root, host)
+            try:
```

Comment:
Can you factor out this cleanup logic into a function? We don't know how long it'll stay around, and it might evolve over time, and we should keep it consistent between the various places.